File: GLAthlete/app/src/main/java/nittcprocon/glathlete/RenderingTask.java
package nittcprocon.glathlete;

class RenderingTask {
    private Model model;
    private ShaderProgram shader;

    RenderingTask(Model model, ShaderProgram shader) {
        model(model).shader(shader);
    }

    void render() {
        model.drawWithShader(shader);
    }

    /*
     * スクストに倣ったGetter/Setterのスタイル
     */

    RenderingTask model(Model model) {
        this.model = model;
        return this;
    }

    RenderingTask shader(ShaderProgram shader) {
        this.shader = shader;
        return this;
    }

    Model model() {
        return model;
    }

    ShaderProgram shader() {
        return shader;
    }
}

File: GLAthlete/app/src/main/java/nittcprocon/glathlete/RenderingTask.kt
package nittcprocon.glathlete

internal class RenderingTask(var model: Model, var shader: ShaderProgram) {
    fun render() {
        model.drawWithShader(shader)
    }

    /*
     * スクストに倣ったGetter/Setterのスタイル
     */

    fun model(model: Model): RenderingTask {
        this.model = model
        return this
    }

    fun shader(shader: ShaderProgram): RenderingTask {
        this.shader = shader
        return this
    }

    fun model(): Model {
        return model
    }

    fun shader(): ShaderProgram {
        return shader
    }
}
